pkg/operator/workloads/api_status.go

```diff
@@ -343,36 +345,27 @@ func getActiveAPIStatus(apiStatuses []*resource.APIStatus, ctx *context.Context)
 	return latestAPIStatus
 }
 
-func apiGroupStatusCode(apiStatuses []*resource.APIStatus, ctx *context.Context) resource.StatusCode {
-	if len(apiStatuses) == 0 {
-		return resource.StatusStopped
-	}
-
+func apiGroupStatusCode(apiStatuses []*resource.APIStatus, groupedReplicaCounts resource.GroupedReplicaCounts, ctx *context.Context) resource.StatusCode {
 	apiName := apiStatuses[0].APIName
 	ctxAPI := ctx.APIs[apiName]
 
-	if ctxAPI == nil {
+	var currentAPIStatus *resource.APIStatus
+	if ctxAPI != nil {
 		for _, apiStatus := range apiStatuses {
-			if apiStatus.TotalReady() > 0 {
-				return resource.StatusStopping
+			if apiStatus.ResourceID == ctxAPI.ID {
+				currentAPIStatus = apiStatus
 			}
 		}
-		return resource.StatusStopped
-	}
-
-	for _, apiStatus := range apiStatuses {
-		if apiStatus.TotalReady() > 0 {
-			return resource.StatusLive
-		}
 	}
 
-	for _, apiStatus := range apiStatuses {
-		if apiStatus.ResourceID == ctxAPI.ID {
-			return apiStatus.Code
+	if ctxAPI == nil || currentAPIStatus == nil || groupedReplicaCounts.Requested == 0 {
```

Comment:
When `groupedReplicaCounts.Requested == 0` be true without `MaxReplicas == 0` being true?